src/librustdoc/core.rs

```diff
@@ -286,7 +286,7 @@ pub(crate) fn create_config(
                 let body = hir.body(hir.body_owned_by(def_id));
                 debug!("visiting body for {def_id:?}");
                 EmitIgnoredResolutionErrors::new(tcx).visit_body(body);
-                (rustc_interface::DEFAULT_QUERY_PROVIDERS.typeck)(tcx, def_id)
+                (rustc_interface::DEFAULT_QUERY_PROVIDERS.0.typeck)(tcx, def_id)
```

Comment:
Would it maybe be better to make `DEFAULT_QUERY_PROVIDERS` have a `struct DefaultQueryProviders { queries: ..., hooks: ... }`? I don't really like `.0` here...

~~wishing we had anon structs 😅~~

Reply:
I also merged the extern providers into that new struct, but I can drop that commit.